common/utils/include/claragenomics/utils/device_buffer.cuh

```diff
@@ -32,26 +31,24 @@ public:
 };
 
 template <typename T>
-class device_storage
+class device_buffer
 {
 public:
     using value_type = T;
-    device_storage() = delete;
-    device_storage(size_t n_elements, int32_t device_id)
+    device_buffer()  = delete;
+    device_buffer(size_t n_elements, int32_t device_id)
```

Comment:
Can we drop `device_id` argument from the constructor? Most use cases have consecutive/multiple calls to `device_buffer` and calling `cudaSetDevice` for each one of them seems unnecessary. 
We should let the caller set the context before calling these methods.

Reply:
Sure, makes sense. Originally, I added it (and kept it as a data member) because I thought you have to ensure you are on the right device before calling cudaFree.
So, you are right, since that's gone there is no added value in having that. Will remove it.